operator/pkg/init.go

```diff
@@ -42,13 +42,14 @@ var (
 
 // InitOptions defines all the init workflow options.
 type InitOptions struct {
-	Name           string
-	Namespace      string
-	Kubeconfig     *rest.Config
-	KarmadaVersion string
-	CRDTarball     operatorv1alpha1.CRDTarball
-	KarmadaDataDir string
-	Karmada        *operatorv1alpha1.Karmada
+	Name                    string
+	Namespace               string
+	Kubeconfig              *rest.Config
+	KarmadaVersion          string
+	CRDTarball              operatorv1alpha1.CRDTarball
+	CustomCertificateConfig operatorv1alpha1.CustomCertificate
```

Comment:
We can consider not adding a new field `CustomCertificateConfig` since `Karmada` already contains the configuration for `customcertificates`.

Reply:
That's a good point. Thanks for bringing it up. I considered that, but if you take a closer look, the root `Karmada` object is not part of this type. It's possible to add it and get rid of a lot of fields here, but that would take a refactor with very wide scope. I think it's fine for now, as this object is meant to carry the only the info needed in downstream functions.

Reply:
> the root Karmada object is not part of this type

Sorry, I didn't get it. Could you please explain it more?

Reply:
We can add the entire `Karmada` object to this field and get rid of some of the other fields, but that will introduce a refactor of a very wide scope with a lot of changes to the code. Can we avoid doing that for now?

Reply:
Ok. Now I see where the confusion is coming from. I could have it explained it a bit better, so it's my fault. This is the [struct](https://github.com/karmada-io/karmada/blob/master/operator/pkg/init.go#L75) I meant to refer to it. It's what's used to carry information to the downstream functions, not `InitOptions`. Notice how the root `Karmada` type is not in there.

Reply:
That struct is then abstracted behind this [interface](https://github.com/karmada-io/karmada/blob/master/operator/pkg/tasks/init/data.go#L28) which is what is used directly in those functions.

Reply:
@zhzhuang-zju , just let me know if have any other comments/questions.

Reply:
hah, Thank you for the clarification. I understand that `CustomCertificateConfig` is required for `initdata`, but is it possible to obtain it through `initoptions.Karmada.Spec.CustomCertificate` instead of adding it directly to `initoptions`? I actually don't want `initoptions` to become too large.

Reply:
> We can add the entire Karmada object to this field and get rid of some of the other fields, but that will introduce a refactor of a very wide scope with a lot of changes to the code. Can we avoid doing that for now?

Yeah, I agree that.